### PR TITLE
Replaces most manual istype checks with the appropriate helpers.

### PR DESCRIPTION
**Pull request by NonQueueingMatt**
Cherry picked from the previous monolith PR.

### CODE REVIEW

**Comment by NonQueueingMatt** [comment]
!review

File: code/game/machinery/bots/mulebot.dm [hunk 1]
```diff
@@ -237,7 +237,7 @@
 		return
 	if (usr.stat)
 		return
-	if ((in_range(src, usr) && istype(src.loc, /turf)) || (istype(usr, /mob/living/silicon)))
+	if ((in_range(src, usr) && issilicon(src.loc, /turf)) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/bots/mulebot.dm:240]: Wrong istypes

File: code/game/machinery/computer/pod.dm [hunk 2]
```diff
@@ -103,7 +103,7 @@
 /obj/machinery/computer/pod/Topic(href, href_list)
 	if(..())
 		return 1
-	if((usr.contents.Find(src) || (in_range(src, usr) && istype(loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if((usr.contents.Find(src) || (in_range(src, usr) && issilicon(loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/computer/pod.dm:106]: ditto

File: code/game/machinery/computer/prisoner.dm [hunk 3]
```diff
@@ -69,7 +69,7 @@
 	Topic(href, href_list)
 		if(..())
 			return
-		if((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+		if((usr.contents.Find(src) || (in_range(src, usr) && issilicon(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/computer/prisoner.dm:72]: ditto

File: code/game/machinery/computer/prisonshuttle.dm [hunk 4]
```diff
@@ -80,7 +80,7 @@ var/prison_shuttle_timeleft = 0
 		if(..())
 			return
 
-		if ((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+		if ((usr.contents.Find(src) || (in_range(src, usr) && issilicon(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/computer/prisonshuttle.dm:83]: ditto

File: code/game/machinery/computer/specops_shuttle.dm [hunk 5]
```diff
@@ -276,7 +276,7 @@ var/specops_shuttle_timeleft = 0
 	if(..())
 		return 1
 
-	if ((usr.contents.Find(src) || (in_range(src, usr) && istype(loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if ((usr.contents.Find(src) || (in_range(src, usr) && issilicon(loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/computer/specops_shuttle.dm:279]: ditto

File: code/game/machinery/computer/syndicate_specops_shuttle.dm [hunk 6]
```diff
@@ -219,7 +219,7 @@ var/syndicate_elite_shuttle_timeleft = 0
 	if(..())
 		return 1
 
-	if ((usr.contents.Find(src) || (in_range(src, usr) && istype(loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if ((usr.contents.Find(src) || (in_range(src, usr) && issilicon(loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/computer/syndicate_specops_shuttle.dm:222]: ditto

File: code/game/machinery/kitchen/gibber.dm [hunk 7]
```diff
@@ -120,7 +120,7 @@
 		to_chat(user, "<span class='danger'>\The [src] is locked and running, wait for it to finish.</span>")
 		return
 
-	if(!(istype(victim, /mob/living/carbon)) && !(istype(victim, /mob/living/simple_animal)) )
+	if(!(iscarbon(victim)) && !(istype(victim, /mob/living/simple_animal)) )
```
Comment (Doxxmedearly) [inline code/game/machinery/kitchen/gibber.dm:123]: I think we have an istype for animals. isanimal?

File: code/game/machinery/machinery.dm [hunk 8]
```diff
@@ -256,13 +256,13 @@ Class Procs:
...
-	if ( ! (istype(usr, /mob/living/carbon/human) || \
-			istype(usr, /mob/living/silicon)))
+	if ( ! (ishuman(usr) || \
+			issilicon(usr)))
 		to_chat(usr, "<span class='warning'>You don't have the dexterity to do this!</span>")
 		return 1
 /*
 	//distance checks are made by atom/proc/DblClick
-	if ((get_dist(src, user) > 1 || !istype(src.loc, /turf)) && !istype(user, /mob/living/silicon))
+	if ((get_dist(src, user) > 1 || !issilicon(src.loc, /turf)) && !istype(user))
```
Comment (Doxxmedearly) [inline code/game/machinery/machinery.dm:265]: not that commented out stuff matters, but wrong proc

File: code/game/machinery/mecha_fabricator.dm [hunk 9]
```diff
@@ -197,8 +197,8 @@
 	else
 		to_chat(user, "The fabricator cannot hold more [sname].")
 
-/obj/machinery/mecha_part_fabricator/MouseDrop_T(mob/living/carbon/human/target as mob, mob/user as mob)
-	if (!istype(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user, /mob/living/silicon/ai))
+/obj/machinery/mecha_part_fabricator/MouseDrop_T(mob/living/carbon/human/target as mob, mob/user)
+	if (!isAI(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user))
```
Comment (Doxxmedearly) [inline code/game/machinery/mecha_fabricator.dm:201]: proc issues

File: code/game/machinery/navbeacon.dm [hunk 10]
```diff
@@ -190,7 +190,7 @@ Transponder Codes:<UL>"}
 		..()
 		if (usr.stat)
 			return
-		if ((in_range(src, usr) && istype(src.loc, /turf)) || (istype(usr, /mob/living/silicon)))
+		if ((in_range(src, usr) && issilicon(src.loc, /turf)) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/navbeacon.dm:193]: ditto

File: code/game/machinery/newscaster.dm [hunk 11]
```diff
@@ -136,7 +136,7 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 	if(!user.IsAdvancedToolUser())
 		return 0
 
-	if(istype(user, /mob/living/carbon/human) || istype(user,/mob/living/silicon) )
+	if(ishuman(user) || istype(user,/mob/living/silicon) )
```
Comment (Doxxmedearly) [inline code/game/machinery/newscaster.dm:139]:
```suggestion
	if(ishuman(user) || issilicon(user) )
```

File: code/game/machinery/newscaster.dm [hunk 12]
```diff
@@ -404,7 +404,7 @@ var/list/obj/machinery/newscaster/allCasters = list() //Global list that will co
 /obj/machinery/newscaster/Topic(href, href_list)
 	if(..())
 		return
-	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && issilicon(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/newscaster.dm:407]: your find/replace-fu is weak

File: code/game/machinery/spaceheater.dm [hunk 13]
```diff
@@ -140,7 +140,7 @@
 /obj/machinery/space_heater/Topic(href, href_list)
 	if (usr.stat)
 		return
-	if ((in_range(src, usr) && istype(src.loc, /turf)) || (istype(usr, /mob/living/silicon)))
+	if ((in_range(src, usr) && issilicon(src.loc, /turf)) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/spaceheater.dm:143]: ditto

File: code/game/machinery/suit_storage_unit.dm [hunk 14]
```diff
@@ -171,7 +171,7 @@
 /obj/machinery/suit_storage_unit/Topic(href, href_list) //I fucking HATE this proc
 	if(..())
 		return
-	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon/ai)))
+	if ((usr.contents.Find(src) || ((get_dist(src, usr) <= 1) && isAI(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/game/machinery/suit_storage_unit.dm:174]: ditto

File: code/game/objects/effects/chem/chemsmoke.dm [hunk 15]
```diff
@@ -177,7 +177,7 @@
 		for(var/turf/T in targetTurfs)
 			chemholder.reagents.touch_turf(T)
 			for(var/atom/A in T.contents)
-				if(istype(A, /obj/effect/effect/smoke/chem) || istype(A, /mob))
+				if(ismob(A, /obj/effect/effect/smoke/chem) || istype(A))
```
Comment (Doxxmedearly) [inline code/game/objects/effects/chem/chemsmoke.dm:180]: ditto

File: code/game/objects/items/stacks/medical.dm [hunk 16]
```diff
@@ -25,17 +25,17 @@ Contains:
 	var/heal_burn = 0
 	var/apply_sounds
 
-/obj/item/stack/medical/attack(mob/living/M as mob, mob/user as mob)
-	if (!istype(M) || istype(M, /mob/living/silicon) || istype(M, /mob/living/simple_animal/spiderbot))
+/obj/item/stack/medical/attack(mob/living/M as mob, mob/user)
+	if (!issilicon(M) || istype(M) || istype(M, /mob/living/simple_animal/spiderbot))
```
Comment (Doxxmedearly) [inline code/game/objects/items/stacks/medical.dm:29]: mix-up with !s and types

File: code/game/objects/items/weapons/implants/implanter.dm [hunk 17]
```diff
@@ -31,8 +31,8 @@
 		src.icon_state = "implanter0"
 	return
 
-/obj/item/implanter/attack(mob/M as mob, mob/user as mob, var/target_zone)
-	if (!istype(M, /mob/living/carbon))
+/obj/item/implanter/attack(mob/M as mob, mob/user, var/target_zone)
```
Comment (Doxxmedearly) [inline code/game/objects/items/weapons/implants/implanter.dm:34]:
```suggestion
/obj/item/implanter/attack(mob/M, mob/user, var/target_zone)
```

File: code/game/objects/items/weapons/storage/boxes.dm [hunk 18]
```diff
@@ -48,10 +48,10 @@
 /obj/item/storage/box/attack_generic(var/mob/user)
 	if(!chewable)
 		return
-	if (istype(user, /mob/living))
+	if (isliving(user))
 		var/mob/living/L = user
 
-		if (istype(L, /mob/living/carbon/alien/diona) || istype(L, /mob/living/simple_animal) || istype(L, /mob/living/carbon/human))//Monkey-like things do attack_generic, not crew
+		if (ishuman(L, /mob/living/carbon/alien/diona) || istype(L, /mob/living/simple_animal) || istype(L))//Monkey-like things do attack_generic, not crew
```
Comment (Doxxmedearly) [inline code/game/objects/items/weapons/storage/boxes.dm:54]: proc mixing

File: code/modules/awaymissions/bluespaceartillery.dm [hunk 19]
```diff
@@ -43,10 +43,10 @@
 /obj/machinery/computer/artillerycontrol/proc/announce_and_fire(var/turf/t, var/mob/user)
 	if(!istype(t))
 		return
-	if ((user.contents.Find(src) || (in_range(src, user) && istype(src.loc, /turf))) || (istype(user, /mob/living/silicon)))
+	if ((user.contents.Find(src) || (in_range(src, user) && issilicon(src.loc, /turf))) || (istype(user)))
 		if (user.stat || user.restrained()) return
 		if (src.reload < 180) return
-		if ((user.contents.Find(src) || (in_range(src, user) && istype(src.loc, /turf))) || (istype(user, /mob/living/silicon)))
+		if ((user.contents.Find(src) || (in_range(src, user) && issilicon(src.loc, /turf))) || (istype(user)))
```
Comment (Doxxmedearly) [inline code/modules/awaymissions/bluespaceartillery.dm:49]: more mismatching

File: code/modules/holodeck/HolodeckControl.dm [hunk 20]
```diff
@@ -104,7 +104,7 @@
 	if(..())
 		return 1
 
-	if((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if((usr.contents.Find(src) || (in_range(src, usr) && issilicon(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/holodeck/HolodeckControl.dm:107]: ditto

File: code/modules/mob/language/synthetic.dm [hunk 21]
```diff
@@ -30,7 +30,7 @@
 
 		if(drone_only && !istype(S,/mob/living/silicon/robot/drone))
 			continue
-		else if(istype(S , /mob/living/silicon/ai))
+		else if(isAI(S ))
```
Comment (Doxxmedearly) [inline code/modules/mob/language/synthetic.dm:33]:
```suggestion
		else if(isAI(S))
```

File: code/modules/mob/living/silicon/robot/drone/drone_console.dm [hunk 22]
```diff
@@ -54,7 +54,7 @@
 		to_chat(usr, SPAN_WARNING("Access denied."))
 		return
 
-	if((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if((usr.contents.Find(src) || (in_range(src, usr) && issilicon(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/mob/living/silicon/robot/drone/drone_console.dm:57]: ditto

File: code/modules/mob/living/simple_animal/constructs/soulstone.dm [hunk 23]
```diff
@@ -15,8 +15,8 @@
 
 //////////////////////////////Capturing////////////////////////////////////////////////////////
 
-/obj/item/device/soulstone/attack(mob/living/carbon/human/M as mob, mob/user as mob)
-	if(!istype(M, /mob/living/carbon/human))//If target is not a human.
+/obj/item/device/soulstone/attack(mob/living/carbon/human/M as mob, mob/user)
```
Comment (Doxxmedearly) [inline code/modules/mob/living/simple_animal/constructs/soulstone.dm:18]:
```suggestion
/obj/item/device/soulstone/attack(mob/living/carbon/human/M, mob/user)
```

File: code/modules/paperwork/clipboard.dm [hunk 24]
```diff
@@ -169,7 +169,7 @@
 
 			if(P && (P.loc == src) && istype(P, /obj/item/paper) )
 
-				if(!(istype(usr, /mob/living/carbon/human) || istype(usr, /mob/abstract/observer) || istype(usr, /mob/living/silicon)))
+				if(!(ishuman(usr) || issilicon(usr, /mob/abstract/observer) || istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/paperwork/clipboard.dm:172]: ditto

File: code/modules/paperwork/folders.dm [hunk 25]
```diff
@@ -79,7 +79,7 @@
 		else if(href_list["read"])
 			var/obj/item/paper/P = locate(href_list["read"])
 			if(P && (P.loc == src) && istype(P))
-				if(!(istype(usr, /mob/living/carbon/human) || istype(usr, /mob/abstract/observer) || istype(usr, /mob/living/silicon)))
+				if(!(ishuman(usr) || issilicon(usr, /mob/abstract/observer) || istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/paperwork/folders.dm:82]: ditto

File: code/modules/paperwork/paper.dm [hunk 26]
```diff
@@ -101,7 +101,7 @@
 
 
 /obj/item/paper/proc/show_content(mob/user, forceshow)
-	var/can_read = (istype(user, /mob/living/carbon/human) || isobserver(user) || istype(user, /mob/living/silicon)) || forceshow
+	var/can_read = (ishuman(user) || isobserver(user) || issilicon(user)) || forceshow
```
Comment (Doxxmedearly) [inline code/modules/paperwork/paper.dm:104]: ditto

File: code/modules/paperwork/paper_bundle.dm [hunk 27]
```diff
@@ -118,7 +118,7 @@
 
 	if(istype(pages[page], /obj/item/paper))
 		var/obj/item/paper/P = W
-		if(!(istype(usr, /mob/living/carbon/human) || istype(usr, /mob/abstract/observer) || istype(usr, /mob/living/silicon)))
+		if(!(ishuman(usr) || issilicon(usr, /mob/abstract/observer) || istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/paperwork/paper_bundle.dm:121]: ditto

File: code/modules/power/turbine.dm [hunk 28]
```diff
@@ -180,11 +180,11 @@
 	if (usr.stat || usr.restrained() )
 		return
 	if (usr.IsAdvancedToolUser())
-		if(!istype(usr, /mob/living/silicon/ai))
+		if(!isAI(usr))
 			to_chat(usr, "<span class='warning'>You don't have the dexterity to do this!</span>")
 			return
 
-	if (( usr.machine==src && ((get_dist(src, usr) <= 1) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon/ai)))
+	if (( usr.machine==src && ((get_dist(src, usr) <= 1) && isAI(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/power/turbine.dm:187]: ditto

File: code/modules/power/turbine.dm [hunk 29]
```diff
@@ -253,7 +253,7 @@
 /obj/machinery/computer/turbine_computer/Topic(href, href_list)
 	if(..())
 		return
-	if ((usr.contents.Find(src) || (in_range(src, usr) && istype(src.loc, /turf))) || (istype(usr, /mob/living/silicon)))
+	if ((usr.contents.Find(src) || (in_range(src, usr) && issilicon(src.loc, /turf))) || (istype(usr)))
```
Comment (Doxxmedearly) [inline code/modules/power/turbine.dm:256]: ditto

File: code/modules/reagents/Chemistry-Machinery.dm [hunk 30]
```diff
@@ -769,8 +769,8 @@
 			if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
 				break
 
-/obj/machinery/reagentgrinder/MouseDrop_T(mob/living/carbon/human/target as mob, mob/user as mob)
-	if (!istype(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user, /mob/living/silicon/ai))
+/obj/machinery/reagentgrinder/MouseDrop_T(mob/living/carbon/human/target as mob, mob/user)
+	if (!isAI(target) || target.buckled || get_dist(user, src) > 1 || get_dist(user, target) > 1 || user.stat || istype(user))
```
Comment (Doxxmedearly) [inline code/modules/reagents/Chemistry-Machinery.dm:773]: ditto

File: code/modules/vehicles/cargo_train.dm [hunk 31]
```diff
@@ -263,7 +263,7 @@
 /obj/vehicle/train/cargo/trolley/load(var/atom/movable/C)
 	if(ismob(C) && !passenger_allowed)
 		return 0
-	if(!istype(C,/obj/machinery) && !istype(C,/obj/structure/closet) && !istype(C,/obj/structure/largecrate) && !istype(C,/obj/structure/reagent_dispensers) && !istype(C,/obj/structure/ore_box) && !istype(C, /mob/living/carbon/human))
+	if(!ishuman(C,/obj/machinery) && !istype(C,/obj/structure/closet) && !istype(C,/obj/structure/largecrate) && !istype(C,/obj/structure/reagent_dispensers) && !istype(C,/obj/structure/ore_box) && !istype(C))
```
Comment (Doxxmedearly) [inline code/modules/vehicles/cargo_train.dm:266]: ditto

**Comment by NonQueueingMatt** [comment]
i will have to reopen this later

**Pull request closed without merging**